migration_challenge_keras_image/Local Notebook.ipynb

```diff
@@ -138,39 +138,67 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "import os\n"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "%%time\n",
+    "%time\n",
+    "\n",
     "!rm -rf data/train\n",
     "!rm -rf data/test\n",
+    "import os\n",
+    "import concurrent.futures\n",
+    "from random import randint\n",
+    "\n",
+    "[os.makedirs(os.path.join(\"data/train\", \"digit-%d\" % ix), exist_ok=True) for ix in range(10)]\n",
+    "[os.makedirs(os.path.join(\"data/test\", \"digit-%d\" % ix), exist_ok=True) for ix in range(10)] \n",
```

Comment:
I had thought this would be negligible before, but just tested this by itself and saw a speedup from 8min 27s to 7min 20s - so agree it's worth doing!

I'd probably prefer a `for` loop though for readability - since the list comprehensions don't have any meaningful results and there are only a small number of iterations (10)?

E.g. could do something like:

```python
for ix in range(10):
    os.makedirs(os.path.join("data/train", "digit-%d" % ix), exist_ok=True)
    os.makedirs(os.path.join("data/test", "digit-%d" % ix), exist_ok=True)
```

Reply:
Replaced with for loop in 59fc64ff88a367f472b73fe52b3e43622bcf4c26